src/UnitfulAstro.jl

```diff
@@ -86,9 +86,76 @@ import UnitfulAngles: arcminute, arcsecond
 # Total electron content unit (used for ionospheric physics and low-frequency radio astronomy)
 @unit TECU       "TECU"     TotalElectronContentUnit  1e16*m^-2                 false
 
+@logscale mag "mag" Magnitude 10 -2.5 false
+
+#https://en.wikipedia.org/wiki/AB_magnitude     
+@logunit    AB_mag  "AB mag"        Magnitude   3631Jy
+
+# Bolometric magnitudes (cf IAU 2015)
+# https://en.wikipedia.org/wiki/Absolute_magnitude#Bolometric_magnitude
+@logunit    bol_Mag "bol Mag"       Magnitude   3.0128e28*W
+@logunit    bol_mag "bol mag"       Magnitude   2.518_021_002e-8*W*m^-2
+
+#https://en.wikipedia.org/wiki/Apparent_magnitude#Standard_reference_values
+#https://www.cfa.harvard.edu/~dfabricant/huchra/ay145/mags.html
+@logunit    U_mag   "Johnson U mag" Magnitude   1810Jy
+@logunit    B_mag   "Johnson B mag" Magnitude   4260Jy
+@logunit    V_mag   "Johnson V mag" Magnitude   3640Jy
+@logunit    R_mag   "Johnson R mag" Magnitude   3080Jy
+@logunit    I_mag   "Johnson I mag" Magnitude   2550Jy
+@logunit    J_mag   "Johnson J mag" Magnitude   1600Jy
+@logunit    H_mag   "Johnson H mag" Magnitude   1080Jy
+@logunit    K_mag   "Johnson K mag" Magnitude    670Jy
+@logunit    g_mag   "Gunn g mag"    Magnitude   3730Jy
+@logunit    r_mag   "Gunn r mag"    Magnitude   4490Jy
+@logunit    i_mag   "Gunn i mag"    Magnitude   4760Jy
+@logunit    z_mag   "Gunn z mag"    Magnitude   4810Jy
+
+Unitful.isrootpower_dim(::typeof(Unitful.dimension(AB_mag))) = false
+
+
+struct MagnitudeError <: Exception
+    x
```

Comment:
Please, do not use [fields with abstract types](https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-fields-with-abstract-type-1).  `String` should be a safe choice for the type of this field

Reply:
This was inspired by [Unitful.jl's AffineError](https://github.com/ajkeller34/Unitful.jl/blob/c01affaa445f0f5307a61cbfe0172d17c45369e9/src/utils.jl#L236)
```jl
"""
    struct AffineError <: Exception
An invalid operation was attempted with affine units / quantities.
"""
struct AffineError <: Exception
    x
end

Base.showerror(io::IO, e::AffineError) = print(io, "AffineError: $(e.x)")
```

Reply:
Surely when throwing error performance isn't a big deal.  Still, the way the field is used (only to print the error message) suggests that this field should really be a string, not else.

Most error types in `Base` have the message filed marked with a loose `AbstractString`.

Reply:
I can live with `<: AbstractString`

Reply:
I was looking at `Base` and I came across [`showerror`](https://github.com/JuliaLang/julia/blob/8c445663547791e59f33f9e8d49b276332107614/base/errorshow.jl).  I feel like `MagnitudeError` should be typed with `<:Exception`.

Reply:
I'm confused, isn't it already an `Exception`? :thinking:

Reply:
Oh, [now I see](https://github.com/JuliaAstro/UnitfulAstro.jl/pull/9/commits/aa6674379475ef757a833008e03ea68e7e49a4ac) what you mean.  When i suggested to use `String` or `AbstractString` I was referring to the field (the thread refers to the line of the field and I always talked about the field), not the error itself, which is of course fine with `Exception` :wink:

Reply:
I was going to change it but then thought about it some more . . .